horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/user/DatabaseUserBackend.java

```diff
@@ -104,43 +93,21 @@ private static UserData toUserInfo(UserInfo info) {
       userInfo.persist();
    }
 
-   private void addTeamMembership(UserInfo userInfo, String teamName, TeamRole role) {
-      Optional<Team> storedTeam = Team.find("teamName", teamName).firstResultOptional();
-      userInfo.teams.add(new TeamMembership(userInfo, storedTeam.orElseGet(() -> Team.getEntityManager().merge(new Team(teamName))), role));
-   }
-
    public static final class NewUserParameterConverter implements Function<Object[], String[]> {
       @Override public String[] apply(Object[] objects) {
-         return new String[] { ((NewUser) objects[0]).user.username };
+         return new String[] { ((UserService.NewUser) objects[0]).user.username };
       }
    }
-
-   @Authenticated
+   private void addTeamMembership(UserInfo userInfo, String teamName, TeamRole role) {
+      Optional<Team> storedTeam = Team.find("teamName", teamName).firstResultOptional();
+      userInfo.teams.add(new TeamMembership(userInfo, storedTeam.orElseGet(() -> Team.getEntityManager().merge(new Team(teamName))), role));
+   }
    @Override public List<String> getTeams() {
       List<Team> teams = Team.listAll();
       return teams.stream().map(t -> t.teamName + "-team").collect(toList());
    }
-
-   @WithRoles(addUsername = true)
```

Comment:
need to keep `@WithRoles` annotation because of row level security

Reply:
This method is still implemented in `UserServiceImpl` and has the `@WithRoles` annotation